payments/models.py

```diff
@@ -154,3 +171,374 @@ def __str__(self):
+
+            # If the lease is being changed
+            self._check_same_lease(old_instance)
+
+    def is_full_season(self):
+        return (
+            self.lease.start_date == calculate_berth_lease_start_date()
+            and self.lease.end_date == calculate_berth_lease_end_date()
+        )
+
+    def _create_order_lines(self, section):
+        for service in ProductServiceType.FIXED_SERVICES():
+            # If the section has the service (prop) and it's valid (True)
+            if hasattr(section, service.name.lower()) and getattr(
+                section, service.name.lower()
+            ):
+                # Retrieve the product for that service
+                product = AdditionalProduct.objects.get(
+                    service=service, period=PeriodType.SEASON
+                )
+
+                OrderLine.objects.create(
+                    order=self, product=product,
+                )
+
+    def _assign_product_from_object_id(self):
+        # Try to get a BerthProduct (BP)
+        product = BerthProduct.objects.filter(id=self._product_object_id).first()
+        # If the BP was not found, try getting a WinterStorageProduct
+        product = (
+            product
+            if product
+            else WinterStorageProduct.objects.filter(id=self._product_object_id).first()
+        )
+        # If for some reason neither was found (shouldn't be the case), raise an error
+        if not product:
+            raise ValidationError(_("The product passed is not valid"))
+
+        self.product = product
+
+    def _assign_lease_from_object_id(self):
+        # Try to get a BerthLease (BL)
+        lease = BerthLease.objects.filter(id=self._lease_object_id).first()
+        # If the BL was not found, try getting a WinterStorageLease
+        lease = (
+            lease
+            if lease
+            else WinterStorageLease.objects.filter(id=self._lease_object_id).first()
+        )
+        # If for some reason neither was found, raise an error
+        if not lease:
+            raise ValidationError(_("The lease passed is not valid"))
+
+        self.lease = lease
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+
+        # If the product is being added from the admin (only the ID is passed)
+        if self._product_object_id and not self.product:
+            self._assign_product_from_object_id()
+
+        # If the lease is being added from the admin (only the ID is passed)
+        if self._lease_object_id and not self.lease:
+            self._assign_lease_from_object_id()
+
+        creating = self._state.adding
+        # If the product instance is being passed
+        # Price has to be assigned before saving but only if creating
+        if creating and self.product:
+            # Assign the price and tax from the product
+            self.price = self.product.price_value
+            self.tax_percentage = self.product.tax_percentage
```

Comment:
so this should be something like: 
```python
if creating and any([not self.price, not self.tax_percentage]):
    price = self.product.price
    tax_percentage = self.product.tax_percentage
    if self.lease and not self.is_full_season():
        price = calculate_lease_partial_price()
    self.price = self.price or price
    self.tax_percentage = self.tax_percentage or tax_percentage
```

Reply:
Would it the

Reply:
```python
if self.lease and not self.is_full_season():
        price = calculate_lease_partial_price()
```
So that would `calculate_product_partial_season_price` with the `berth/ws product` as "base" price?

Reply:
yeah, berth/ws products are restricted to season, so just calculate based on that